File: FrontEnd/lib/domain/dtos/auth/login_request_dto.dart
/// @file login_request_dto.dart
/// @description DTO de entrada para el caso de uso de login.
/// @module Core
/// @layer Domain
library;

// ═══════════════════════════════════════════════════════════════════════════════
// LOGIN REQUEST DTO
// ═══════════════════════════════════════════════════════════════════════════════

/// DTO de login. Usado por el ViewModel y el use case de autenticación.
class LoginRequestDto {
  const LoginRequestDto({
    required this.email,
    required this.password,
  });

  /// Email del usuario.
  final String email;

  /// Contraseña en texto plano.
  final String password;
}

File: BackEnd/src/domain/dtos/auth/login-request.dto.ts
/**
 * @file login-request.dto.ts
 * @description DTO de solicitud de login de usuario.
 * @module Auth
 * @layer Domain
 */

import { IsEmail, IsString, MinLength } from 'class-validator';

/**
 * DTO de login. Validado en el controlador antes de llegar al use case.
 */
export class LoginRequestDTO {
  /** Email del usuario */
  @IsEmail()
  email!: string;

  /** Contraseña en texto plano */
  @IsString()
  @MinLength(1)
  password!: string;
}
